### PR TITLE
Decouple model from database

### PR DESCRIPTION
**Pull request by msal4**
#### Summary
Updates PullRequest fields to use `*type` instead of using sql nullable types.

#### Ticket Link

JIRA: https://mattermost.atlassian.net/browse/MM-26814
Fixes [#16975](https://github.com/mattermost/mattermost-server/issues/16975)

### CODE REVIEW

File: server/cherry_pick.go [hunk 1]
```diff
@@ -92,7 +92,7 @@ func (s *Server) handleCherryPick(ctx context.Context, commenter, body string, p
 	command := getCommand(body)
 	args := strings.Split(command, " ")
 	mlog.Info("Args", mlog.String("Args", body))
-	if !pr.Merged.Valid || !pr.Merged.Bool {
+	if pr.Merged == nil || !*pr.Merged {
```
Comment (agnivade) [inline server/cherry_pick.go:95]: Can we have helper methods for each of these fields? Like `IsMergedFalse`, `IsMilestoneNumberEmpty` which wraps the nil check and the false/0 check.

**Review by agnivade — CHANGES_REQUESTED** [review]

File: server/cherry_pick.go (hunk 1 above)
Comment (msal4) [reply server/cherry_pick.go:95]: I'll name them `GetFieldName()` similar to `github.PullRequest` e.g. `GetMerged()`

**Review by agnivade — APPROVED** [review]
Looks so clean! Amazing work!

**Comment by msal4** [comment]
@agnivade Thanks :)

**Review by isacikgoz — APPROVED** [review]

File: server/cherry_pick.go [hunk 2]
```diff
@@ -153,8 +150,8 @@ func (s *Server) checkIfNeedCherryPick(pr *model.PullRequest) {
 	prLabels := labelsToStringArray(labels)
 	for _, prLabel := range prLabels {
 		if prLabel == "CherryPick/Approved" {
-			milestoneNumber := int(pr.MilestoneNumber.Int64)
-			milestone := getMilestone(pr.MilestoneTitle.String)
+			milestoneNumber := int(*pr.MilestoneNumber)
```
Comment (isacikgoz) [inline server/cherry_pick.go:153]:
```suggestion
			milestoneNumber := int(pr.GetMilestoneNumber())
```

File: model/pull_request.go [hunk 3]
```diff
@@ -28,9 +27,37 @@ type PullRequest struct {
...
 	MergeCommitSHA      string `db:"-"`
-	MaintainerCanModify sql.NullBool
-	MilestoneNumber     sql.NullInt64
-	MilestoneTitle      sql.NullString
+	MaintainerCanModify *bool
+	MilestoneNumber     *int64
+	MilestoneTitle      *string
+}
+
+func (pr *PullRequest) GetMerged() bool {
```
Comment (isacikgoz) [inline model/pull_request.go:37]: Not a blocking thing but, I'd prefer Go convention:
```suggestion
func (pr *PullRequest) Merged() bool {
```

**Review by isacikgoz — APPROVED** [review]
Just a couple of non blocking points.

File: model/pull_request.go (hunk 3 above)
Comment (msal4) [reply model/pull_request.go:37]: @isacikgoz Wouldn't that conflict with the field name?

Comment (agnivade) [reply model/pull_request.go:37]: Ahh yeah .. that's true, the field is exported. Well, we could unexport and have it accessible only via a getter. But it's out of scope for this PR.

Comment (msal4) [reply model/pull_request.go:37]: I'll happily do a separate PR addressing it if I get a green light.

Comment (agnivade) [reply model/pull_request.go:37]: I don't think that's necessary, just for the sake of sticking to a convention. A comment explaining why the "Get" part is needed should suffice.

**Pull request merged**